picoquic/util.c

```diff
@@ -164,6 +164,35 @@ int debug_printf_reset(int suspended)
     return ret;
 }
 
+int picoquic_sprintf(char* buf, size_t buf_len, const char* fmt, ...)
+{
+    va_list args;
+    va_start(args, fmt);
+    int res = vsnprintf(buf, buf_len, fmt, args);
+    va_end(args);
+
+    // vsnprintf returns <0 for errors and >=0 for nb of characters required.
+    // We return 0 when printing was successful.
+    return res >= 0 ? (res >= buf_len) : res;
+}
+
```

Comment:
With this, you can remove a bunch of remove a bunch of `#ifdef` from the code, and that is very good. But the problem with `vsnprintf` is the same as with `sprintf`. According to https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/vsnprintf-vsnprintf-vsnprintf-l-vsnwprintf-vsnwprintf-l?view=vs-2019, _More secure versions of these functions are available; see vsnprintf_s, _vsnprintf_s, _vsnprintf_s_l, _vsnwprintf_s, _vsnwprintf_s_l._

That means you still need an `#ifdef _WINDOWS` here, to use the secure version on Windows.

Reply:
OK.